ext/js/dom/document-util.js

```diff
@@ -113,21 +113,32 @@ export class DocumentUtil {
         const text = source.text();
         const textLength = text.length;
         const textEndAnchor = textLength - endLength;
-        let pos1 = startLength;
-        let pos2 = textEndAnchor;
+
+        /**
+         * Cursor representing the relative position of the sentence.
+         * @namespace
+         */
+        const cursor = {
```

Comment:
I'd recommend not creating an object and just using `cursorStart` and `cursorEnd`. Object notation can have different performance characteristics than inline variables.

Reply:
To provide a bit more description on this, the reason why I'm suggesting this change is because the `cursor` namespace is more cosmetic vs providing semantic utility. And since there at the very least _can_ be a performance difference with objects, esp. with the variables being used as iteration variables, I recommended to simplify.

For some additional context, I can also list a few situations where I _would_ find such an object to be useful:
* It's passed as a parameter to another function
* It's returned
* It's nullable
* It's already a pre-existing class which closely matches the use case
  (example for something unrelated, imagine a `new Point(x, y)`)